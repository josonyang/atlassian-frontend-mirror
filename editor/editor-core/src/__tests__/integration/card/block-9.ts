import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  getDocFromElement,
  editable,
} from '@atlaskit/editor-test-helpers/integration/helpers';
import {
  goToEditorTestingWDExample,
  mountEditor,
} from '@atlaskit/editor-test-helpers/testing-example-page';
import { ConfluenceCardProvider } from '@atlaskit/editor-test-helpers/confluence-card-provider';
import blockCardDatasourceAdf from './_fixtures_/block-card-datasource.adf.json';

type ClientType = Parameters<typeof goToEditorTestingWDExample>[0];

BrowserTestCase(
  'card: block card with datasource should copy and paste multiple datasource tables',
  {},
  async (client: ClientType, testName: string) => {
    const page = await goToEditorTestingWDExample(client);

    const cardProviderPromise = Promise.resolve(
      new ConfluenceCardProvider('prod'),
    );

    await mountEditor(
      page,
      {
        appearance: 'full-page',
        allowTextAlignment: true,
        defaultValue: JSON.stringify(blockCardDatasourceAdf),
        smartLinks: {
          provider: cardProviderPromise,
          allowBlockCards: true,
          allowDatasource: true,
        },
      },
      undefined,
      { clickInEditor: false },
    );

    await page.click('.datasourceView-content-wrap');

    await page.keys(['ArrowDown', 'Enter']);

    await page.selectAll();

    await page.copy();

    await page.click('p');

    await page.paste();

    expect(
      await page.$eval(editable, getDocFromElement),
    ).toMatchCustomDocSnapshot(testName);
  },
);
