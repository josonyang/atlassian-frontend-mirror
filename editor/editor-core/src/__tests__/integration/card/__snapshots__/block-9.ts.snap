// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card: block card with datasource should copy and paste multiple datasource tables 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "data": null,
        "datasource": Object {
          "id": "some-datasource-id",
          "parameters": Object {
            "cloudId": "DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b",
            "jql": "project=EDM",
          },
          "views": Array [
            Object {
              "properties": Object {
                "columns": Array [
                  Object {
                    "key": "type",
                  },
                  Object {
                    "key": "summary",
                  },
                  Object {
                    "key": "assignee",
                  },
                ],
              },
              "type": "table",
            },
          ],
        },
        "layout": "center",
        "url": null,
        "width": null,
      },
      "type": "blockCard",
    },
    Object {
      "type": "paragraph",
    },
    Object {
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "data": null,
        "datasource": Object {
          "id": "some-datasource-id",
          "parameters": Object {
            "cloudId": "DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b",
            "jql": "project=EDM",
          },
          "views": Array [
            Object {
              "properties": Object {
                "columns": Array [
                  Object {
                    "key": "type",
                  },
                  Object {
                    "key": "summary",
                  },
                  Object {
                    "key": "assignee",
                  },
                ],
              },
              "type": "table",
            },
          ],
        },
        "layout": null,
        "url": null,
        "width": null,
      },
      "type": "blockCard",
    },
    Object {
      "type": "paragraph",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
